### ISSUE TITLE
Bug in PERL 5.6 ?

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#4801]
Migrated from [rt.perl.org#8198](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=8198) (status was 'resolved')

Searchable as RT8198$

**Comment by p5pRT** [comment]
### From SHAHK2@Nationwide.com

Dear Sir&#8203;:

Hi \! My name is Kirit Shah and I work for Nationwide Insurance\, Columbus\,
Ohio\. We are migrating our PERL from 5\.004 to 5\.6 version\. I have a problem
with defining local variable in sub\-routines\. I don't know if this is a bug
or I don't understand correctly\. But this program works perfect in PERL
5\.004 version\. Let me explain you what is my code and where I am getting
problem\.\.

\-\-\-\-\-\-\-\-\-\-\-\-\-begining of PERL program
\#\!/vol/perl/5\.6/bin/perl
\.\.\.\.
\.\.\.
\.\.\.
\.\.\.
\.\.\.
\# there is a function called here\.\.

if \(&check\_for\_error\($p1\, $p2\, $p3\) \{
&nbsp;          return "Error";
\} else \{
&nbsp;          return
\}

\.\.\.
\.\.\.
\.\.\.
\# here is function
sub check\_for\_error \{
local\($p1\, $p2\, $p3\) = @&#8203;\_;


sub check\_for\_error \{
&nbsp;        local\($p1\, $p2\, $p3\) = @&#8203;\_;
&nbsp;        if\($p3 eq "D"\) \{
&nbsp;                local\($table\) = "dept";
&nbsp;        \} elsif\($p3 eq "C"\) \{
&nbsp;                local\($table\) = "emp";
&nbsp;        \} else \{
&nbsp;                &app\_abend\(undef\, "invalid parm "\);
&nbsp;        \}
&nbsp;        local\($sql\) = \<\<"END\_OF\_SQL";
&nbsp;        select  date
&nbsp;        from    $table
&nbsp;        where   date = '$p2'
&nbsp;        and     fund\_seq\_id = $p1
END\_OF\_SQL
&nbsp;        local\($csr\) = &ora\_open\($lda\,$sql\)
&nbsp;                || &app\_abend\(undef\, "Can't open cursor to check :
$ora\_errstr"\);
&nbsp;        local\($date\) = &ora\_fetch\($csr\);
&nbsp;        &ora\_close\($csr\);
&nbsp;        if\($date\) \{
&nbsp;                return 1;
&nbsp;        \} else \{
&nbsp;                return undef;
&nbsp;        \}
\}


---
When I run this program I am getting an oracle error that table doesnot
exists\. But when I kept "Print $table" statement  as follows&#8203;:



\-\-\-\-\-\-\-\-\-\-\-\-\-begining of perl program
\#\!/vol/perl/5\.6/bin/perl
\.\.\.\.
\.\.\.
\.\.\.
\.\.\.
\.\.\.
\# there is a function called here\.\.

if \(&check\_for\_error\($p1\, $p2\, $p3\) \{
&nbsp;          return "Error";
\} else \{
&nbsp;          return
\}

\.\.\.
\.\.\.
\.\.\.
\# here is function
sub check\_for\_error \{
local\($p1\, $p2\, $p3\) = @&#8203;\_;


sub check\_for\_error \{
&nbsp;        local\($p1\, $p2\, $p3\) = @&#8203;\_;
&nbsp;        if\($p3 eq "D"\) \{
\# added for debugging purpose
&nbsp;        print "\\n this is a D case \\n"\);
&nbsp;                local\($table\) = "dept";
&nbsp;        \} elsif\($p3 eq "C"\) \{
\# added for debugging purpose
&nbsp;        print "\\n this is a D case \\n"\);
&nbsp;                local\($table\) = "emp";
&nbsp;        \} else \{
&nbsp;       \# added for debugging purpose
&nbsp;        print "\\n error case \\n"\);
&nbsp;                &app\_abend\(undef\, "invalid parm "\);
&nbsp;        \}
\# added for debugging purpose
&nbsp;        print "\\n Table Name is : $table : \\n"\);
&nbsp;        local\($sql\) = \<\<"END\_OF\_SQL";
&nbsp;        select  date
&nbsp;        from    $table
&nbsp;        where   date = '$p2'
&nbsp;        and     fund\_seq\_id = $p1
END\_OF\_SQL
&nbsp;        local\($csr\) = &ora\_open\($lda\,$sql\)
&nbsp;                || &app\_abend\(undef\, "Can't open cursor to check :
$ora\_errstr"\);
&nbsp;        local\($date\) = &ora\_fetch\($csr\);
&nbsp;        &ora\_close\($csr\);
&nbsp;        if\($date\) \{
&nbsp;                return 1;
&nbsp;        \} else \{
&nbsp;                return undef;
&nbsp;        \}
\}


---
It printed as follows&#8203;:
 this is a D case
 Table Name is :  :

Can you explain is this a problem or my mistake ? This version works
perfectly in Perl 5\.004 version\. I am using Perl from Unix\.

Should you need any more information\, please let me know\.

Thanks\!


Kirit P Shah
614\-677\-8825
shahk2@&#8203;nationwide\.com

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

Kirit Shah wrote&#8203;:
> Hi \! My name is Kirit Shah and I work for Nationwide 
> Insurance\, Columbus\,
> Ohio\. We are migrating our PERL from 5\.004 to 5\.6 version\. I 
> have a problem
> with defining local variable in sub\-routines\. I don't know if 
> this is a bug
> or I don't understand correctly\. But this program works 
> perfect in PERL
> 5\.004 version\. Let me explain you what is my code and where I 
> am getting
> problem\.\.

I find it strange that this worked under 5\.004\.\.\.  In any case\, I 
don't believe this is a perl bug\, but a problem with your script\.

The problem is\, I believe\, that you have "local\($table\)" in each
branch of your if statement\.

The local statement only takes effect for the enclosing block\, so
as the if branch ends\, "$table" returns to its previous \(unset\) value\.

At a minimum\, you should do this&#8203;:

sub check\_for\_error \{
&nbsp;        local\($p1\, $p2\, $p3\) = @&#8203;\_;
&nbsp;      \# this is now the $table that the if's will modify
&nbsp;      local $table;
&nbsp;        if\($p3 eq "D"\) \{
&nbsp;              \# no local\!
&nbsp;                $table = "dept";
&nbsp;        \} elsif\($p3 eq "C"\) \{
&nbsp;              \# no local\!
&nbsp;                $table = "emp";
&nbsp;        \} else \{
&nbsp;                &app\_abend\(undef\, "invalid parm "\);
&nbsp;        \}
&nbsp;        local\($sql\) = \<\<"END\_OF\_SQL";
&nbsp;        select  date
&nbsp;        from    $table
&nbsp;        where   date = '$p2'
&nbsp;        and     fund\_seq\_id = $p1
END\_OF\_SQL
&nbsp;        local\($csr\) = &ora\_open\($lda\,$sql\)
&nbsp;                || &app\_abend\(undef\, "Can't open cursor to check :
$ora\_errstr"\);
&nbsp;        local\($date\) = &ora\_fetch\($csr\);
&nbsp;        &ora\_close\($csr\);
&nbsp;        if\($date\) \{
&nbsp;                return 1;
&nbsp;        \} else \{
&nbsp;                return undef;
&nbsp;        \}
\}

You should \_really\_ be using "my" instead of "local"\.

sub check\_for\_error \{
&nbsp;        my \($p1\, $p2\, $p3\) = @&#8203;\_;
&nbsp;      my $table;
&nbsp;        if\($p3 eq "D"\) \{
&nbsp;                $table = "dept";
&nbsp;        \} elsif\($p3 eq "C"\) \{
&nbsp;                $table = "emp";
&nbsp;        \} else \{
&nbsp;                &app\_abend\(undef\, "invalid parm "\);
&nbsp;        \}
&nbsp;        my $sql = \<\<"END\_OF\_SQL";
&nbsp;        select  date
&nbsp;        from    $table
&nbsp;        where   date = '$p2'
&nbsp;        and     fund\_seq\_id = $p1
END\_OF\_SQL
&nbsp;        my \($csr\) = &ora\_open\($lda\,$sql\)
&nbsp;                || &app\_abend\(undef\, "Can't open cursor to check :
$ora\_errstr"\);
&nbsp;        my \($date\) = &ora\_fetch\($csr\);
&nbsp;        &ora\_close\($csr\);
&nbsp;        if\($date\) \{
&nbsp;                return 1;
&nbsp;        \} else \{
&nbsp;                return undef;
&nbsp;        \}
\}

Please check out perlfaq7 for more details about why "my" is 
closer to what you need\, by doing " perldoc perlfaq7" or
"perldoc \-q local" and reading the perlfaq7 answer\.

Hope this helps\!
\-\- 
Mike


\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*
Bear Stearns is not responsible for any recommendation\, solicitation\, 
offer or agreement or any information about any transaction\, customer 
account or account activity contained in this communication\.
\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

Doug / Mike / Jim&#8203;:

Thank you very much for your help\. I could fix my problem with your help
and hints\. I'm sorry for directly sending this not to 'perlbug@&#8203;perl\.com'
ID\.  Even though this is a logical error\, it is a really strange this the
same function works good with Perl 5\.004\.

Thank you again\.

Kirit P Shah
\-\-\-\-\- Forwarded by Kirit Shah/Nationwide/NWIE on 01/09/02 04&#8203;:19 PM \-\-\-\-\-
&nbsp;                                                                                                                
&nbsp;                       "Wilson\, Doug"                                                                           
&nbsp;                       \<Doug\_Wilson@&#8203;intuit  To&#8203;: "'SHAHK2@&#8203;Nationwide\.com'" \<SHAHK2@&#8203;Nationwide\.com>               
&nbsp;                       \.com>                cc&#8203;:                                                                 
&nbsp;                                            bcc&#8203;:                                                                
&nbsp;                                            Subject&#8203;:                           RE&#8203;: \[ID 20020109\.003\] Bug in     
&nbsp;                                            PERL 5\.6 ?                                                          
&nbsp;                                                                                                                
&nbsp;                       01/09/02 11&#8203;:59 AM                                                                        
&nbsp;                                                                                                                
&nbsp;                                                                                                                







> From&#8203;: SHAHK2@&#8203;Nationwide\.com \[mailto&#8203;:SHAHK2@&#8203;Nationwide\.com\]
> \.\.\.
> \# here is function
> sub check\_for\_error \{
> local\($p1\, $p2\, $p3\) = @&#8203;\_;
>
>
> sub check\_for\_error \{
>         local\($p1\, $p2\, $p3\) = @&#8203;\_;
>         if\($p3 eq "D"\) \{
>                 local\($table\) = "dept";
>         \} elsif\($p3 eq "C"\) \{
>                 local\($table\) = "emp";
>         \} else \{


It looks like you need some basic lessons in perl 5\.
my\(\) has replaced local\(\) for most uses in perl 5\.
and $table will have to be declared outside of that
'if' block because the way you have it\, e\.g\. $table
is only set to 'dept' within the if block\, its only
set to 'emp' within the elsif block\, etc\. and returns
to its original value after the entire if block\.


If you are migrating\, an important thing to do on any
script over a few lines is to use strict\. Yes\, it'll
generate alot of errors to start\, but after fixing them
you'll end up with more maintainable code\.


This is way off topic for p5p\, so I'm just replying to
you and not the list\. p5p is for the development of
perl itself\. If you need help\, try perlmonks\.org\,
and search for things like 'use strict'\, 'use strict or die'
\(bad syntax but a good node\)\, and 'use strict and warnings'\.
I'm also available on a $$/hr\. basis :\-\)


HTH\,
Douglas Wilson
\-\-\-\-\- Forwarded by Kirit Shah/Nationwide/NWIE on 01/09/02 04&#8203;:19 PM \-\-\-\-\-
&nbsp;                                                                                                                
&nbsp;                       "Giroux\, Mike                                                                            
&nbsp;                       \(Exchange\)"          To&#8203;: "'SHAHK2@&#8203;Nationwide\.com'" \<SHAHK2@&#8203;Nationwide\.com>               
&nbsp;                       \<mgiroux@&#8203;bear\.com>   cc&#8203;: "'perl5\-porters@&#8203;perl\.org'" \<perl5\-porters@&#8203;perl\.org>             
&nbsp;                                            bcc&#8203;:                                                                
&nbsp;                                            Subject&#8203;:                           RE&#8203;: \[ID 20020109\.003\] Bug in     
&nbsp;                                            PERL 5\.6 ?                                                          
&nbsp;                                                                                                                
&nbsp;                       01/09/02 11&#8203;:58 AM                                                                        
&nbsp;                                                                                                                
&nbsp;                                                                                                                




Kirit Shah wrote&#8203;:
> Hi \! My name is Kirit Shah and I work for Nationwide
> Insurance\, Columbus\,
> Ohio\. We are migrating our PERL from 5\.004 to 5\.6 version\. I
> have a problem
> with defining local variable in sub\-routines\. I don't know if
> this is a bug
> or I don't understand correctly\. But this program works
> perfect in PERL
> 5\.004 version\. Let me explain you what is my code and where I
> am getting
> problem\.\.

I find it strange that this worked under 5\.004\.\.\.  In any case\, I
don't believe this is a perl bug\, but a problem with your script\.

The problem is\, I believe\, that you have "local\($table\)" in each
branch of your if statement\.

The local statement only takes effect for the enclosing block\, so
as the if branch ends\, "$table" returns to its previous \(unset\) value\.

At a minimum\, you should do this&#8203;:

sub check\_for\_error \{
&nbsp;        local\($p1\, $p2\, $p3\) = @&#8203;\_;
&nbsp;             \# this is now the $table that the if's will modify
&nbsp;             local $table;
&nbsp;        if\($p3 eq "D"\) \{
&nbsp;                     \# no local\!
&nbsp;                $table = "dept";
&nbsp;        \} elsif\($p3 eq "C"\) \{
&nbsp;                     \# no local\!
&nbsp;                $table = "emp";
&nbsp;        \} else \{
&nbsp;                &app\_abend\(undef\, "invalid parm "\);
&nbsp;        \}
&nbsp;        local\($sql\) = \<\<"END\_OF\_SQL";
&nbsp;        select  date
&nbsp;        from    $table
&nbsp;        where   date = '$p2'
&nbsp;        and     fund\_seq\_id = $p1
END\_OF\_SQL
&nbsp;        local\($csr\) = &ora\_open\($lda\,$sql\)
&nbsp;                || &app\_abend\(undef\, "Can't open cursor to check :
$ora\_errstr"\);
&nbsp;        local\($date\) = &ora\_fetch\($csr\);
&nbsp;        &ora\_close\($csr\);
&nbsp;        if\($date\) \{
&nbsp;                return 1;
&nbsp;        \} else \{
&nbsp;                return undef;
&nbsp;        \}
\}

You should \_really\_ be using "my" instead of "local"\.

sub check\_for\_error \{
&nbsp;        my \($p1\, $p2\, $p3\) = @&#8203;\_;
&nbsp;             my $table;
&nbsp;        if\($p3 eq "D"\) \{
&nbsp;                $table = "dept";
&nbsp;        \} elsif\($p3 eq "C"\) \{
&nbsp;                $table = "emp";
&nbsp;        \} else \{
&nbsp;                &app\_abend\(undef\, "invalid parm "\);
&nbsp;        \}
&nbsp;        my $sql = \<\<"END\_OF\_SQL";
&nbsp;        select  date
&nbsp;        from    $table
&nbsp;        where   date = '$p2'
&nbsp;        and     fund\_seq\_id = $p1
END\_OF\_SQL
&nbsp;        my \($csr\) = &ora\_open\($lda\,$sql\)
&nbsp;                || &app\_abend\(undef\, "Can't open cursor to check :
$ora\_errstr"\);
&nbsp;        my \($date\) = &ora\_fetch\($csr\);
&nbsp;        &ora\_close\($csr\);
&nbsp;        if\($date\) \{
&nbsp;                return 1;
&nbsp;        \} else \{
&nbsp;                return undef;
&nbsp;        \}
\}

Please check out perlfaq7 for more details about why "my" is
closer to what you need\, by doing " perldoc perlfaq7" or
"perldoc \-q local" and reading the perlfaq7 answer\.

Hope this helps\!
\-\-
Mike


\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*
Bear Stearns is not responsible for any recommendation\, solicitation\,
offer or agreement or any information about any transaction\, customer
account or account activity contained in this communication\.
\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*


\-\-\-\-\- Forwarded by Kirit Shah/Nationwide/NWIE on 01/09/02 04&#8203;:19 PM \-\-\-\-\-
&nbsp;                                                                                                                
&nbsp;                       jim cromie                                                                               
&nbsp;                       \<jcromie@&#8203;divsol\.com  To&#8203;: SHAHK2@&#8203;Nationwide\.com                                           
&nbsp;                       >                    cc&#8203;: perl5\-porters\-help@&#8203;perl\.\.org                                     
&nbsp;                                            bcc&#8203;:                                                                
&nbsp;                                            Subject&#8203;:                           Re&#8203;: \[ID 20020109\.003\] Bug in     
&nbsp;                                            PERL 5\.6 ? \- nope                                                   
&nbsp;                                                                                                                
&nbsp;                       01/09/02 03&#8203;:06 PM                                                                        
&nbsp;                                                                                                                
&nbsp;                                                                                                                




SHAHK2@&#8203;Nationwide\.com wrote&#8203;:

>Dear Sir&#8203;:
>
>Hi \! My name is Kirit Shah and I work for Nationwide Insurance\, Columbus\,
>Ohio\. We are migrating our PERL from 5\.004 to 5\.6 version\. I have a
problem
>with defining local variable in sub\-routines\. I don't know if this is a
bug
>or I don't understand correctly\. But this program works perfect in PERL
>5\.004 version\. Let me explain you what is my code and where I am getting
>problem\.\.
>
short answer \-

generally\, you should be using my\, not local\, which doesnt mean quite
what you think it does\.

also\, youre correct to question your understanding\,
 comp\.lang\.perl\.misc usenet is a better place for this kind of question\.\.

you can always come here afterwards if  responses on other lists suggest
that it is a bug\.


keeep in mind that my is like automatic variables in C\, they disappear once
code execution gets outside of the braces which the declaration exists in\.

in other words\, in

sub func \{
&nbsp;    my \($var\, $var2\) =@&#8203;\_;
\}


vars are temporary inside the function\, and go away outside\.

this also applies to

>sub check\_for\_error \{
>\.\.\.
>

>if\($p3 eq "D"\) \{
>                local\($table\) = "dept";
>        \} elsif\($p3 eq "C"\) \{
>                local\($table\) = "emp";
>

those locals cant just be converted to mys\, as $table will ceace to exist
outside the if\-block\.

put it above the if instead\.

my $table;
if \(\.\.\.\) \{\}



modulo these problems\, you could start with something like

find \. \-name '\*\.p\[lm\]' \-exec 'perl \-pi\.bak \-e "s/\\blocal\\b/my/"

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

Dear Sir&#8203;:

Hi \! My name is Kirit Shah and I work for Nationwide Insurance\, Columbus\,
Ohio\. We are migrating our PERL from 5\.004 to 5\.6 version\. I have a problem
with defining local variable in sub\-routines\. I don't know if this is a bug
or I don't understand correctly\. But this program works perfect in PERL
5\.004 version\. Let me explain you what is my code and where I am getting
problem\.\.

\-\-\-\-\-\-\-\-\-\-\-\-\-begining of PERL program
\#\!/vol/perl/5\.6/bin/perl
\.\.\.\.
\.\.\.
\.\.\.
\.\.\.
\.\.\.
\# there is a function called here\.\.

if \(&check\_for\_error\($p1\, $p2\, $p3\) \{
&nbsp;          return "Error";
\} else \{
&nbsp;          return
\}

\.\.\.
\.\.\.
\.\.\.
\# here is function
sub check\_for\_error \{
local\($p1\, $p2\, $p3\) = @&#8203;\_;


sub check\_for\_error \{
&nbsp;        local\($p1\, $p2\, $p3\) = @&#8203;\_;
&nbsp;        if\($p3 eq "D"\) \{
&nbsp;                local\($table\) = "dept";
&nbsp;        \} elsif\($p3 eq "C"\) \{
&nbsp;                local\($table\) = "emp";
&nbsp;        \} else \{
&nbsp;                &app\_abend\(undef\, "invalid parm "\);
&nbsp;        \}
&nbsp;        local\($sql\) = \<\<"END\_OF\_SQL";
&nbsp;        select  date
&nbsp;        from    $table
&nbsp;        where   date = '$p2'
&nbsp;        and     fund\_seq\_id = $p1
END\_OF\_SQL
&nbsp;        local\($csr\) = &ora\_open\($lda\,$sql\)
&nbsp;                || &app\_abend\(undef\, "Can't open cursor to check :
$ora\_errstr"\);
&nbsp;        local\($date\) = &ora\_fetch\($csr\);
&nbsp;        &ora\_close\($csr\);
&nbsp;        if\($date\) \{
&nbsp;                return 1;
&nbsp;        \} else \{
&nbsp;                return undef;
&nbsp;        \}
\}


---
When I run this program I am getting an oracle error that table doesnot
exists\. But when I kept "Print $table" statement  as follows&#8203;:



\-\-\-\-\-\-\-\-\-\-\-\-\-begining of perl program
\#\!/vol/perl/5\.6/bin/perl
\.\.\.\.
\.\.\.
\.\.\.
\.\.\.
\.\.\.
\# there is a function called here\.\.

if \(&check\_for\_error\($p1\, $p2\, $p3\) \{
&nbsp;          return "Error";
\} else \{
&nbsp;          return
\}

\.\.\.
\.\.\.
\.\.\.
\# here is function
sub check\_for\_error \{
local\($p1\, $p2\, $p3\) = @&#8203;\_;


sub check\_for\_error \{
&nbsp;        local\($p1\, $p2\, $p3\) = @&#8203;\_;
&nbsp;        if\($p3 eq "D"\) \{
\# added for debugging purpose
&nbsp;        print "\\n this is a D case \\n"\);
&nbsp;                local\($table\) = "dept";
&nbsp;        \} elsif\($p3 eq "C"\) \{
\# added for debugging purpose
&nbsp;        print "\\n this is a D case \\n"\);
&nbsp;                local\($table\) = "emp";
&nbsp;        \} else \{
&nbsp;       \# added for debugging purpose
&nbsp;        print "\\n error case \\n"\);
&nbsp;                &app\_abend\(undef\, "invalid parm "\);
&nbsp;        \}
\# added for debugging purpose
&nbsp;        print "\\n Table Name is : $table : \\n"\);
&nbsp;        local\($sql\) = \<\<"END\_OF\_SQL";
&nbsp;        select  date
&nbsp;        from    $table
&nbsp;        where   date = '$p2'
&nbsp;        and     fund\_seq\_id = $p1
END\_OF\_SQL
&nbsp;        local\($csr\) = &ora\_open\($lda\,$sql\)
&nbsp;                || &app\_abend\(undef\, "Can't open cursor to check :
$ora\_errstr"\);
&nbsp;        local\($date\) = &ora\_fetch\($csr\);
&nbsp;        &ora\_close\($csr\);
&nbsp;        if\($date\) \{
&nbsp;                return 1;
&nbsp;        \} else \{
&nbsp;                return undef;
&nbsp;        \}
\}


---
It printed as follows&#8203;:
 this is a D case
 Table Name is :  :

Can you explain is this a problem or my mistake ? This version works
perfectly in Perl 5\.004 version\. I am using Perl from Unix\.

Should you need any more information\, please let me know\.

Thanks\!


Kirit P Shah
614\-677\-8825
shahk2@&#8203;nationwide\.com
